test/widgets/drive_detail_page_test.dart

```diff
@@ -0,0 +1,164 @@
+        start: 'Waypoint',
+        status: RideStatus.approved,
+      ));
+
+      when(processor.processUrl(any)).thenReturn(jsonEncode(drive.toJsonForApi()));
+
+      await pumpMaterial(tester, DriveDetailPage.fromDrive(drive));
+
+      // Wait for the drive to be fully loaded
+      await tester.pump();
+
+      expect(find.text(drive.start), findsNWidgets(2));
+      expect(find.text('Waypoint'), findsOneWidget);
+    });
+
+    testWidgets('Can handle duplicate waypoints', (WidgetTester tester) async {
+      String waypoint = 'Waypoint';
+      drive.rides!.addAll(List.generate(
+        3,
+        (index) => RideFactory().generateFake(
+          start: index % 2 == 0 ? waypoint : null,
+          end: index % 2 == 0 ? null : waypoint,
+          status: RideStatus.approved,
+        ),
+      ));
+
+      when(processor.processUrl(any)).thenReturn(jsonEncode(drive.toJsonForApi()));
+
+      await pumpMaterial(tester, DriveDetailPage.fromDrive(drive));
+
+      // Wait for the drive to be fully loaded
+      await tester.pump();
+
+      expect(find.text(drive.start), findsNWidgets(2));
+      expect(find.text(waypoint), findsOneWidget);
+    });
+
+    testWidgets('Can cancel drive', (WidgetTester tester) async {
+      when(processor.processUrl(any)).thenReturn(jsonEncode(drive.toJsonForApi()));
+
+      await pumpMaterial(tester, DriveDetailPage.fromDrive(drive));
+
+      // Wait for the drive to be fully loaded
+      await tester.pump();
+
+      final Finder cancelDriveButton = find.byKey(const Key('cancelDriveButton'));
+      expect(cancelDriveButton, findsOneWidget);
+      await tester.scrollUntilVisible(cancelDriveButton, 500.0);
+      await tester.tap(cancelDriveButton);
+      await tester.pumpAndSettle();
+
+      final Finder cancelDriveYesButton = find.byKey(const Key('cancelDriveYesButton'));
+      expect(cancelDriveYesButton, findsOneWidget);
+      await tester.tap(cancelDriveYesButton);
+      await tester.pumpAndSettle();
+
+      // Verify that the drive was cancelled (but no way to verify body right now)
+      verify(processor.processUrl('/rest/v1/drives?id=eq.${drive.id}')).called(1);
+
+      expect(find.byKey(const Key("cancelledDriveBanner")), findsOneWidget);
+    });
+
+    testWidgets('Can abort cancelling drive', (WidgetTester tester) async {
+      when(processor.processUrl(any)).thenReturn(jsonEncode(drive.toJsonForApi()));
+
+      await pumpMaterial(tester, DriveDetailPage.fromDrive(drive));
+
+      // Wait for the drive to be fully loaded
+      await tester.pump();
+
+      final Finder cancelDriveButton = find.byKey(const Key('cancelDriveButton'));
+      expect(cancelDriveButton, findsOneWidget);
+      await tester.scrollUntilVisible(cancelDriveButton, 500.0);
+      await tester.tap(cancelDriveButton);
+      await tester.pumpAndSettle();
+
+      final Finder cancelDriveNoButton = find.byKey(const Key('cancelDriveNoButton'));
+      expect(cancelDriveNoButton, findsOneWidget);
```

Comment:
shouldn't we also check somewhere if the message in the Dialog is there?

Reply:
Tbh there are a million things we could check, but if there is Yes/No then we can be pretty sure that the dialog is working correctly...